mne/viz/_3d.py

```diff
@@ -1381,6 +1381,188 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
+    """
+    # Import here to avoid circular imports
+    import surfer
+    from surfer import Brain, TimeViewer
+    from ..source_estimate import VectorSourceEstimate
+
+    if not isinstance(stc, VectorSourceEstimate):
+        raise ValueError('stc has to be a vector source estimate')
+
+    subjects_dir = get_subjects_dir(subjects_dir=subjects_dir,
+                                    raise_error=True)
+    subject = _check_subject(stc.subject, subject, True)
+    surfer_version = LooseVersion(surfer.__version__)
+    v08 = LooseVersion('0.8')
+    if surfer_version < v08:
+        raise ImportError("This function requires PySurfer 0.8 (you are "
+                          "running version %s). You can update PySurfer "
+                          "using:\n\n    $ pip install -U pysurfer" %
+                          surfer.__version__)
+
+    if hemi not in ['lh', 'rh', 'split', 'both']:
+        raise ValueError('hemi has to be either "lh", "rh", "split", '
+                         'or "both"')
+
+    time_label, times = _handle_time(time_label, time_unit, stc.times)
+
+    # convert control points to locations in colormap
+    scale_pts, colormap = _limits_to_control_points(clim, stc.data, colormap)
+    transparent = True if transparent is None else transparent
+
+    if hemi in ['both', 'split']:
+        hemis = ['lh', 'rh']
+    else:
+        hemis = [hemi]
+
+    if alpha is None:
+        alpha = brain_alpha
+
+    title = subject if len(hemis) > 1 else '%s - %s' % (subject, hemis[0])
+    with warnings.catch_warnings(record=True):  # traits warnings
+        brain = Brain(subject, hemi=hemi, surf='white', curv=True,
+                      title=title, cortex=cortex, size=size,
+                      background=background, foreground=foreground,
+                      figure=figure, subjects_dir=subjects_dir,
+                      views=views, alpha=brain_alpha)
+
+    for hemi in hemis:
+        hemi_idx = 0 if hemi == 'lh' else 1
+        if hemi_idx == 0:
+            data = stc.data[:len(stc.vertices[0])]
+        else:
+            data = stc.data[len(stc.vertices[0]):]
+        vertices = stc.vertices[hemi_idx]
+        if len(data) > 0:
+            with warnings.catch_warnings(record=True):  # traits warnings
+                brain.add_data(data, colormap=colormap, vertices=vertices,
+                               smoothing_steps=smoothing_steps, time=times,
+                               time_label=time_label, alpha=alpha, hemi=hemi,
+                               colorbar=colorbar, vector_alpha=vector_alpha,
+                               scale_factor=scale_factor,
+                               initial_time=initial_time)
+
+        # scale colormap and set time (index) to display
+        brain.scale_data_colormap(fmin=scale_pts[0], fmid=scale_pts[1],
+                                  fmax=scale_pts[2], transparent=transparent)
+
+    if time_viewer:
+        TimeViewer(brain)
+
+    return brain
```

Comment:
>  There are too many parameters that work for one and not the other and too many branches

This seems like a good time to do refactoring into private functions for mayavi- vs matplotlib-based plotting. This should decrease the branching into more easily digested bits, no?

> it requires PySurfer v0.8

This is just one conditional, right? `if data.ndim == 3 and _check_whatever ...: raise RuntimeError`

> different default values for brain_alpha and alpha, etc

We could expose `brain_alpha=None`, `alpha=None`, and `vector_alpha=1.`, and triage `brain_alpha` and `alpha` to `1.` for standard data and `0.5` for vector data. This doesn't seem so horrible.

I'm sure there are other things that would need to be triaged, but hopefully a single conditional within the PySurfer private function would do it. In my mind we have one big split (whether we use MPL or PySurfer) and one small split (parameter options for PySurfer based on source estimate type). If you also see it this way, it shouldn't be so bad to make the code reflect this, right/